python/lib/sift_py/grpc/_retry.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import json
+from typing import Dict, List, TypedDict
+
+from grpc import StatusCode
+from typing_extensions import Self
+
+
+class RetryPolicy:
+    """
+    Retry policy meant to be used for `sift_py.grpc.transport.SiftChannel`. Users may have the ability to configure their own
+    custom retry policy in the future, but for now this is primarily intended for internal use.
+
+    - [Retry policy schema](https://github.com/grpc/grpc-proto/blob/ec30f589e2519d595688b9a42f88a91bdd6b733f/grpc/service_config/service_config.proto#L136)
+    - [Enable gRPC retry option](https://github.com/grpc/grpc/blob/9a5fdfc3d3a7fc575a394360be4532ee09a85620/include/grpc/impl/channel_arg_names.h#L311)
+    - [Service config option](https://github.com/grpc/grpc/blob/9a5fdfc3d3a7fc575a394360be4532ee09a85620/include/grpc/impl/channel_arg_names.h#L207)
+    """
+
+    config: RetryConfig
+
+    DEFAULT_POLICY: RetryConfig = {
+        "methodConfig": [
+            {
+                # We can configure this on a per-service and RPC basis but for now we'll
+                # apply this across all services and RPCs.
+                "name": [{}],
+                "retryPolicy": {
+                    # gRPC does not allow more than 5 attempts
+                    "maxAttempts": 5,
+                    "initialBackoff": "0.1s",
+                    "maxBackoff": "1s",
+                    "backoffMultiplier": 2,
```

Comment:
With a max of 5 retries we'll do:
0.1
0.2
0.4
0.8
1.0

for a total amount of time between first failure and last attempt of 2.5s by my count. I might bump this so we start earlier but also up a bit faster to give things more time to recover before that last one?

Like: 0.05s and 4x with a max of 5s:
0.05
0.20
0.80
3.2s
5s

which should give more like 9.25s before the final one, while still trying quickly for the first two.

Reply:
yeah that sounds better. just updated